### DISCUSSION TITLE
适配新版粉丝勋章亲密度机制 开发计划

### DISCUSSION THREAD

**Discussion by andywang425** [andywang425/BLTH#520] [Development]
涉及三个功能：点赞，发弹幕，还有观看直播。下面列一些我和群友想到的一些需求：

# 点赞
因为只有给开播的直播间点赞才能拿到亲密度，目前脚本（v7.1.6）只给已开播的直播间点赞。

- 以后可以考虑在给已开播的直播间点完赞后保持一个监听的状态，等其余直播间开播后去点赞。

# 发弹幕

很多用户反映说在已开播的直播间刷屏（发10弹幕）不太好，影响直播间氛围而且容易被禁言。下面是我想到的一些改进措施：

- 允许用户自己设置一个弹幕发送的间隔。有需求的用户可以把发送间隔拉大，从而避免短时间内在同一个直播间发大量弹幕。
- 区别对待未开播的直播间和已开播的直播间。对于未开播的直播间我们可以快速发完10条弹幕，而在已开播的直播间以较慢速度发。

# 观看直播

机制尚不明确。我在 https://github.com/andywang425/BLTH/issues/518#issuecomment-2245544598 提到的机制好像也不太对，具体机制还需进一步挖掘。目前看来这个功能不那么重要，因为根据官方的说法，如果仅仅是完成观看直播任务，粉丝勋章也是会熄灭的。

# 其他

有些用户只是想保持亮着的勋章不熄灭，不在乎加多少亲密度，因此我们也可以为此单独设计一套逻辑，具体怎么做我也没想好。

------------------------------------------

如果有想法欢迎来讨论。有能力的话也可以提个PR。

**Comment by ADJazzzz** [comment]
~~关于观看直播那里，现在改成只有开着直播的房间才能增加亲密度了
目前勋章保活的话只有发弹幕是比较高效的，这个是不管有没有开直播都是增加亲密度的~~
哈哈回调了@andywang425 
![Screenshot_2024-08-02-17-13-33-884_tv.danmaku.bili-edit.jpg](https://github.com/user-attachments/assets/4d1deb57-479d-4fde-850a-056c5c51361b)

**Reply by ADJazzzz** [reply]
一个想法：正在直播的房间也许发表情弹幕可能没那么突兀

**Reply by andywang425** [reply]
> 关于观看直播那里，现在改成只有开着直播的房间才能增加亲密度了

是吗，感谢测试。

> 一个想法：正在直播的房间也许发表情弹幕可能没那么突兀

嗯，我也觉得。如果要支持复杂表情，得用另一种API调用格式，而且还需要写逻辑让用户选择想发送的表情。
![image](https://github.com/user-attachments/assets/8e222fb5-274e-4185-84ac-bc91544884d5)
简单表情（最左边那个）其实就是文字，目前脚本也支持。

**Reply by ADJazzzz** [reply]
我觉得就用最左边那个的就可以，因为目的是把这10个弹幕的刷屏不要太显眼🫠

**Comment by Raiden-1** [comment]
对于只是想保持亮着的勋章不熄灭，不在乎加多少亲密度的，我建议出个选项：对在开播的直播间点赞50~55次（不发弹幕），对未开播的直播间快速发完10条弹幕（因为点赞无效）。这样开脚本立刻就能做完。

**Reply by Raiden-1** [reply]
希望尽快更新出 发送弹幕（10条）-仅对未开播的直播间生效 子选项

**Comment by andywang425** [comment]
最近因为用来开发Vue项目的vscode拓展Vue - Official问题非常严重，开始写代码后分分钟出bug，很折磨。所以暂时先放缓BLTH的开发，等拓展更新一下再继续。

![image](https://github.com/user-attachments/assets/d34f09b6-ea9b-4603-ae8f-d3356668e8a9)

**Reply by andywang425** [reply]
接着这个机会去试了下用IDEA（以及WebStorm）写Vue，发现还挺好用。虽然也存在一些小毛病但是比现在的vscode+volar体验好太多了

**Comment by andywang425** [comment]
众所周知B站从去年就开始给部分API加[WBI签名](https://github.com/SocialSisterYi/bilibili-API-collect/blob/master/docs/misc/sign/wbi.md)，但都是些主站的API。最近我观察到直播区的一部分API也开始上签名了，虽然目前不带加密参数也能用，但很有可能以后是必须要有签名的。
![image](https://github.com/user-attachments/assets/fbc7161f-30b4-44cd-ada9-e1c6daac7966)

因此TODO+1：支持 WBI 签名

**Comment by Only-Mio** [comment]
发弹幕，可以按轮发，每轮所有直播间各发一条，这样就不会刷屏了

**Comment by andywang425** [comment]
又有新改动，计划调整，https://github.com/andywang425/BLTH/discussions/528

- 观看直播任务改成一个一个观看，因为即使同时观看所有直播间也只能给其中一个加亲密度
- 点亮勋章，支持两种方式，发一条弹幕和点赞30次
